### PR TITLE
Runner case name should be case insensitive

### PR DESCRIPTION
**Pull request by uraniumanchor**
# Contributing to the Donation Tracker

- [X] I've added tests or modified existing tests for the change.
- [X] I've humanly end-to-end tested the change by running an instance of the tracker.

### Issue from Pivotal Tracker

https://www.pivotaltracker.com/story/show/169991967

### Description of the Change

Runner names being case insensitive has caused some issues in the past, especially with importing the final schedule.

This enforces case insensitivity at the model level (doing it from the database would lock us to PostgreSQL, far as I can tell, so that's probably out for now). This extends to the api, mostly for free.

### Possible Drawbacks

If two runners really did have the same name but different capitalization... but that was already a theoretical problem before, and this doesn't make that any worse.

### Verification Process

Added tests, tried to add a runner via both the api and the admin pages, and the checks worked.

### CODE REVIEW

File: models/event.py [hunk 1]
```diff
@@ -541,18 +540,40 @@ def get_by_natural_key(self, name):
...
     class Meta:
         app_label = 'tracker'
 
     objects = _Manager()
-    name = models.CharField(max_length=64, unique=True)
+    name = models.CharField(
+        max_length=64,
+        unique=True,
+        error_messages={
+            'unique': 'Runner with this case-insensitive Name already exists.'
```
Comment (faultyserver) [inline models/event.py]: Is this a message that will get shown publicly to end users? Or is it just internal? If it's public, I don't think `case-insensitive` needs to be there. Very few services out there have case sensitivity, so just seeing "name already exists" effectively implies that it's case insensitive.

Comment (uraniumanchor) [reply models/event.py]: If by 'end user' you mean 'somebody trying to add a runner', then yeah, it's public. It's what the admin/api shows.

Comment (faultyserver) [reply models/event.py]: okay, but it's not like actual public as in random people visiting a site and trying to sign themselves up. This seems fine as is then.

Comment (uraniumanchor) [reply models/event.py]: There are definitely some user (as in the general public) facing things that could stand this sort of treatment (I've lost count of the number of times I have to explain to people that the login is case sensitive, dammit Django), but that's a story for another time.

**Pull request merged**